### PR TITLE
feat: use built in `node:test` runner in all template test implementations

### LINKED ISSUE #632: Use Node.js built-in test runner in templates
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

In same spirit to https://github.com/fastify/fastify-cli/issues/631, use the Node.js built-in test runner instead of tap, they are mostly compatibles and would remove an extra dependency.

### Motivation

Use standard built-in modules and reduce dependencies.

### Example

_No response_

### PR DESCRIPTION
**Pull request by big-kahuna-burger**
## Closes #632 #680

For all 4 variants, `[cjs,esm] x [js, ts]` I've used a locally linked cli to generate a project and have `npm i && npm test` pass.
Typescript variants have coverage enabled by default, cjs one using `c8`, esm one using  `--experimental-test-coverage` flag. tsx has been added to devDeps, in order to support ts-esm variant.

## Todos
There are two pending issues with this PR afaik:
- [x] `test/templates/app-ts.test.ts` is skipped instead of letting it break the complete suite. [See more here](https://github.com/fastify/fastify-cli/actions/runs/7001183252/job/19048968732?pr=681#step:5:1127) . Any help would be nice as to what to do with it.
- [x] ~~script `unit:ts-cjs` is set to always pass due to `||`. This is a chicken-egg problem. Here a template test suite is being run, but that test suite also requires a `fastify-cli/helper.js` by name. So, I can't really install that as devDependency of self. I'm sure there is some trick to handle this. Any help appreciated again.~~

- [x] Skip running template suites on ~~14/16~~ <20

- [x] Node 20 on windows glob error

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Fdawgs) [inline .github/workflows/ci.yml:15]: This is a breaking change, as v4 drops CI testing for node 14 and 16.

Comment (big-kahuna-burger) [reply .github/workflows/ci.yml]: Implemented skipping test suites so it exits with 0 on older node versions.
Other cli related tests are still being ran on those.

**Comment by big-kahuna-burger** [comment]
imho this can be reviewed now by maintainers.
[All workflows](https://github.com/big-kahuna-burger/fastify-cli/actions/runs/7017225362) are passing on my fork.

@mcollina @Fdawgs 
PTAL

**Comment by big-kahuna-burger** [comment]
Converted to draft. Waiting for #681 to be merged so this can be rebased on top of it.

**Comment by big-kahuna-burger** [comment]
Good for review again @mcollina.
Rebased on top of master, removed del-cli usage in favor of `--noEmit` flag given to `tsc`

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Could you remove node v14/v16 from CI? The code that this generates would never run there.

cc @Uzlopak @Fdawgs I'm not sure if we have a parameter for this

**Comment by Fdawgs** [comment]
> cc @Uzlopak @Fdawgs I'm not sure if we have a parameter for this

Shared workflow just needs to be updated to use v4, which dropped support for 14 and 16.

**Comment by mcollina** [comment]
I think we removd the v4 tag for now to avoid massive updates everywhere (by auto merge) that were not necessary.

**Comment by Kae7in** [comment]
When is this going to get merged? I got a similar error to #800 when starting a project via `fastify generate example --lang=ts --esm`

**Comment by Kae7in** [comment]
Thank you @mcollina!! That worked. 😄

**Pull request merged**
